src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -117,19 +117,19 @@ protected virtual void Execute(Transaction tx, BlockExecutionContext blCtx, ITxT
 
             if (opts == ExecutionOptions.Commit || opts == ExecutionOptions.None)
             {
-                decimal gasPrice = (decimal)effectiveGasPrice / 1_000_000_000m;
+                float gasPrice = (float)((double)effectiveGasPrice / 1_000_000_000.0);
```

Comment:
is there any benefit of converting to double first? Float should be sufficient.

Reply:
`UInt256` has no direct cast to float

Reply:
Feel free to add one :D